packages/gatsby/src/schema/schema.js

```diff
@@ -968,12 +969,9 @@ const addConvenienceChildrenFields = ({ schemaComposer }) => {
       )
       return
     }
-    children.forEach((many, child) => {
-      if (many) {
-        typeComposer.addFields(createChildrenField(child.getTypeName()))
-      } else {
-        typeComposer.addFields(createChildField(child.getTypeName()))
-      }
+    children.forEach(child => {
+      typeComposer.addFields(createChildrenField(child.getTypeName()))
+      typeComposer.addFields(createChildField(child.getTypeName()))
```

Comment:
That's basically the essence of this PR. Other changed files are just tests/docs updates.